### PR TITLE
fix(esm): cjs to esm as default

### LINKED ISSUE #100: [Feature]: add publint
### **Package Scope**

- [x] Add to an existing package
- [ ] New package

Package name:

### LINKED ISSUE #126: [Feature]: esm first
### **Package Scope**

- [x] Add to an existing package

Package name: @suspensive/react @suspensive/react-query

### PR DESCRIPTION
**Pull request by manudeli**
fix #100 fix #126

# ESM first

We can check our package.json's problem in [publint's website](https://publint.dev/@suspensive/react@1.12.0)
image

1. I followed [publint's guide](https://publint.dev/rules#exports_types_should_be_first)(used by @tanstack/query). and ci will check this rules
image

2. I followed [TypeScript guide](https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing)

## PR Checklist

- [x] I have written documents and tests, if needed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   },
   "license": "MIT",
   "packageManager": "pnpm@8.5.1",
+  "type": "module",
```
Comment (manudeli) [inline package.json:17]: ESM as default

File: packages/react-query/package.json [hunk 2]
```diff
@@ -14,16 +14,21 @@
...
   "sideEffects": false,
+  "type": "module",
   "exports": {
     ".": {
-      "require": "./dist/index.js",
-      "import": "./dist/index.mjs",
-      "types": "./dist/index.d.ts"
+      "import": {
+        "types": "./dist/index.d.ts",
+        "default": "./dist/index.js"
```
Comment (manudeli) [inline packages/react-query/package.json:22]: index.js is esm from now. we don't have to use prefix m(esm)-js.

**Review by manudeli — COMMENTED** [review]
I left my intentions

**Review by okinawaa — APPROVED** [review]
good!

**Pull request merged**